### PR TITLE
Generate string interpolation expression + test

### PR DESCRIPTION
**Pull request by AnaCoda**

### CODE REVIEW

**Review by j3parker — APPROVED** [review]

File: src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs [hunk 1]
```diff
@@ -458,6 +461,17 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 		);
 	}
 
+	private SyntaxList<InterpolatedStringContentSyntax> TransformInterpolations( SyntaxList<InterpolatedStringContentSyntax> interSyntaxs ) {
+		SyntaxList <InterpolatedStringContentSyntax> newInterSyntaxs = interSyntaxs;
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs:465]: `newInterSyntaxs` isn't really adding anything here, right? `interSyntaxs = interSyntaxs.Replace( interSyntaxs[i]` would be equivalent?

File: src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs [hunk 2]
```diff
@@ -458,6 +461,17 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 		);
 	}
 
+	private SyntaxList<InterpolatedStringContentSyntax> TransformInterpolations( SyntaxList<InterpolatedStringContentSyntax> interSyntaxs ) {
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs:464]: The plural of Syntax is Syntaxes (with an _e_)

File: src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs (hunk 1 above)
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs:465]: nit: whitespace

```suggestion
		SyntaxList<InterpolatedStringContentSyntax> newInterSyntaxs = interSyntaxs;
```

Comment (AnaCoda) [reply src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs:465]: There was an issue where the first replace would change the syntax location of the later ones in interSyntaxs, meaning they could no longer be found in the SyntaxTree causing an exception on a type check somewhere else. Using newInterSyntaxs solves that

**Review by omsmith — APPROVED** [review]

**Pull request merged**